src/plugins/ec.ts

```diff
@@ -172,6 +173,33 @@ export class ECPlugin extends BasePlugin {
             }, {});
     }
 
+    private convertProductPriceToNumber(product: Product) {
+        if (product.price) {
+            product.price = this.tryConvertStringPriceToNumber(product.price);
+        }
+
+        return product;
+    }
+
+    private convertImpressionPriceToNumber(impression: BaseImpression) {
+        if (impression.price) {
+            impression.price = this.tryConvertStringPriceToNumber(impression.price);
+        }
+
+        return impression;
+    }
+
+    private tryConvertStringPriceToNumber(price: Number | string): Number | string {
+        if (typeof price === 'string') {
+            let parsedPrice = Number(price.replace(/[^0-9\.-]/g, ''));
+            if (parsedPrice) {
+                return parsedPrice;
+            }
+        }
+
+        return price;
```

Comment:
nit: we could remove the nesting a bit by flipping the condition into a guard clause
```suggestion
        if (typeof price === 'number') {
            return price
        }
        
        return Number(price.replace(/[^0-9\.-]/g, '')) || price;
```